packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1199,29 +1288,51 @@ export class Cluster extends ClusterBase {
 
     // do not create a masters role if one is not provided. Trusting the accountRootPrincipal() is too permissive.
     if (props.mastersRole) {
+      const mastersRole = props.mastersRole;
       this.grantAccess('mastersRoleAccess', props.mastersRole.roleArn, [
         AccessPolicy.fromAccessPolicyName('AmazonEKSClusterAdminPolicy', {
           accessScopeType: AccessScopeType.CLUSTER,
         }),
       ]);
+
+      commonCommandOptions.push(`--role-arn ${mastersRole.roleArn}`);
     }
 
     if (props.albController) {
       this.albController = AlbController.create(this, { ...props.albController, cluster: this });
     }
 
-    // allocate default capacity if non-zero (or default).
-    const minCapacity = props.defaultCapacity ?? DEFAULT_CAPACITY_COUNT;
-    if (minCapacity > 0) {
-      const instanceType = props.defaultCapacityInstance || DEFAULT_CAPACITY_TYPE;
-      this.defaultCapacity = props.defaultCapacityType === DefaultCapacityType.EC2 ?
-        this.addAutoScalingGroupCapacity('DefaultCapacity', { instanceType, minCapacity }) : undefined;
+    // if any of defaultCapacity* properties are set, we need a default capacity(nodegroup)
+    if (props.defaultCapacity !== undefined ||
+        props.defaultCapacityType !== undefined ||
+        props.defaultCapacityInstance !== undefined) {
```

Comment:
nit: if `defaultCapacityType` is auto mode, it doesn't need to enter this condition.